packages/digit_dss/lib/pages/dashboard/dashboard.dart

```diff
@@ -0,0 +1,158 @@
+import 'package:auto_route/auto_route.dart';
+import 'package:digit_components/digit_components.dart';
+import 'package:digit_components/models/digit_table_model.dart';
+import 'package:digit_components/widgets/atoms/digit_toaster.dart';
+import 'package:digit_dss/blocs/dashboard.dart';
+import 'package:digit_dss/widgets/back_navigation_help_header.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+import '../../utils/i18_key_constants.dart' as i18;
+import '../../utils/utils.dart';
+import '../../widgets/dashboard/dashboard_metric_card.dart';
+import '../../widgets/localized.dart';
+
+@RoutePage()
+class UserDashboardPage extends LocalizedStatefulWidget {
+  const UserDashboardPage({
+    super.key,
+  });
+
+  @override
+  State<StatefulWidget> createState() {
+    return UserDashboardPageState();
+  }
+}
+
+class UserDashboardPageState extends LocalizedState<UserDashboardPage> {
+  @override
+  void initState() {
+    context.read<DashboardBloc>().add(DashboardRefreshEvent(
+          projectId: DashboardSingleton().projectId,
+          syncFromServer: false,
+          selectedDate: DateTime.now(),
+        ));
+
+    super.initState();
+  }
+
+  bool isLoading = false;
```

Comment:
remove loading variable since you have a loading state

Reply:
Its necessary to handle the loading pop up, on data load, and refresh, and error state, If its not handled in listener state then the pop up keeps on adding to the navigator stack on refresh